### ISSUE TITLE
Don't trigger unused_optional_binding when assigning to underscore

### ISSUE THREAD

**Issue by Bouke** [realm/SwiftLint#2230]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
[...]/DNS/Tests/DNSTests/DNSTests.swift:130:16: warning: Unused Optional Binding Violation: Prefer `!= nil` over `let _ =` (unused_optional_binding)
```

### Environment

* SwiftLint version: 0.25.1
* Installation method used: homebrew
* Which Xcode version are you using: 9.4
* Do you have a sample that shows the issue?

```bash
echo "if let _ = try? Message(deserialize: data) { }" | swiftlint lint --no-cache --use-stdin
```
```swift
// This triggers a violation:
if let _ = try? Message(deserialize: data) { ... }
```

The problem with this rule is that `_` is already telling Swift that the variable will not be used. I can't seem to find an authoritive answer on this, but it's common convention to assign to `_` to get around `warn_unused_result`.  So this line of code should IMO not trigger the violation.

**Comment by marcelofabri** [comment]
This is exactly that the rule should catch 😉 

If you take a look at the [docs](https://github.com/realm/SwiftLint/blob/master/Rules.md#unused-optional-binding), you'll see that the description is `Prefer != nil over let _ =`.

So the rule wants you to write `if try? Message(deserialize: data) != nil { }`. 

Of course you can also disable this rule if you don't agree with it.

**Comment by SDGGiesbrecht** [comment]
@marcelofabri, is generally correct.

I will only add that unless something has changed in Swift 4, `try?` confuses things with operator precedence, so in your case, you need extra parentheses too. More details in #1432 starting [here](https://github.com/realm/SwiftLint/issues/1432#issuecomment-296396135).

It also looks like the rule can be configured with `ignore_optional_try`. Maybe that is closer to what you want, @Bouke?

**Comment by Bouke** [comment]
The compile warning due to the missing parenthesis did confuse me at first. And (see also the linked thread), it's also a matter preference whether one prefers one over the other:

```swift
if (try? Message(deserialize: data)) != nil { ... }
```
```
if let _ = try? Message(deserialize: data) { ... }
```

So thanks to you both, I'll probably set `ignore_optional_try` and revert some changes in my code.

**Comment by jazminebarroga** [comment]
When `ignore_optional_try` is set to true, the if-let syntax still triggers a violation. However, when using the guard statement, it does not. Is there a reason why the if-let syntax is not covered when `ignore_optional_try` is set to true?
